source/common/router/router.cc

```diff
@@ -2129,10 +2129,14 @@ void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or
   }
   // Check whether we need to send the load report to the LRS or invoke the ORCA
   // callbacks.
-  auto host = upstream_request.upstreamHost();
-  const bool need_to_send_load_report =
-      (host != nullptr) && cluster_->lrsReportMetricNames().has_value();
-  if (!need_to_send_load_report && orca_load_report_callbacks_.expired()) {
+  OptRef<const Upstream::HostDescription> upstream_host = upstream_request.upstreamHost();
+  if (!upstream_host.has_value()) {
+    return;
+  }
+
+  OptRef<Upstream::HostLbPolicyData> host_lb_policy_data = upstream_host->lbPolicyData();
+
+  if (!cluster_->lrsReportMetricNames().has_value() && !host_lb_policy_data.has_value()) {
```

Comment:
Please add a log-line in the if.

I'm trying to wrap my head around the relation between `orca_load_report_callbacks_.expired()` and the current host not having a policy data.
I think that your suggested condition makes more sense, but just not 100% this is covering the same conditions as before.

Reply:
they are basically same condition, except at the case where the lb is gone but the host is still not be destroied. That's a corner case where we won't promise anything.